tests/pkg_fatfs/main.c

```diff
@@ -0,0 +1,314 @@
+                    GiB available\n", fr_gib_i, fr_gib_f, to_gib_i, to_gib_f);
+        }
+    }
+    else {
+        printf("[FAILED]\n");
+        switch (mountresu) {
+            case FR_NO_FILESYSTEM:
+                printf("no filesystem -> you need to format the card to FAT\n");
+                break;
+            case FR_DISK_ERR:
+                printf("error in the low-level disk driver (sdcard_spi)!\n");
+                break;
+            default:
+                printf("error %d -> look ff.h of fatfs package for \
+                        further details\n", mountresu);
+        }
+        return -1;
+    }
+    return 0;
+}
+
+static int _mk(int argc, char **argv)
+{
+    FIL fd;
+
+    if (argc != 2) {
+        printf("usage: %s <filename>\n", argv[0]);
+        return -1;
+    }
+
+    if (f_open(&fd, argv[1], FA_WRITE | FA_CREATE_ALWAYS) == FR_OK) {
+        FRESULT close_resu = f_close(&fd);
+        if (close_resu == FR_OK) {
+            printf("[OK]\n");
+            return 0;
+        }
+        else {
+            printf("[FAILED] (f_close error %d)\n", close_resu);
+            return -2;
+        }
+    }
+    else {
+        printf("failed to create file '%s'\n", argv[1]);
+    }
+    return 0;
+}
+
+static int _read(int argc, char **argv)
+{
+    FIL fd;
+    int resu = 0;
+
+    if (argc != 2) {
+        printf("usage: %s <filename> [<len>]\n", argv[0]);
+        return -1;
+    }
+
+    char buffer[TEST_FATFS_READ_BUFFER_SIZE];
+
+    FRESULT open_resu = f_open(&fd, argv[1], FA_READ | FA_OPEN_EXISTING);
+    if (open_resu == FR_OK) {
+        UINT read_chunk;
+
+        for (uint32_t read = 0; read < (f_size(&fd)); read += read_chunk) {
+            uint32_t to_read = (f_size(&fd)) - read;
+
+            if (to_read > sizeof(buffer)) {
+                to_read = sizeof(buffer);
+            }
+
+            FRESULT lseek_resu = f_lseek(&fd, read);
+            if (lseek_resu != FR_OK) {
+                printf("[FAILED] f_lseek error %d\n", lseek_resu);
+                resu = -3;
+                break;
+            }
+
+            FRESULT read_resu = f_read(&fd, buffer, to_read, &read_chunk);
+
+            if (read_resu != FR_OK) {
+                printf("[FAILED] (f_read error %d)\n", read_resu);
+                resu = -4;
+                break;
+            }
+
+            for (uint32_t i = 0; i < read_chunk; i++) {
+                printf("%c", buffer[i]);
+            }
+        }
+        printf("\n");
+
+        FRESULT close_resu = f_close(&fd);
+        if (close_resu == FR_OK) {
+            printf("[OK]\n");
+            resu = 0;
+        }
+        else {
+            printf("[FAILED] (f_close error %d)\n", open_resu);
+            resu = -5;
+        }
+
+    }
+    else {
+        printf("[FAILED] (f_open error %d)\n", open_resu);
+        resu = -2;
+    }
+
+    return resu;
+}
+
+static int _write(int argc, char **argv)
```

Comment:
does not append as suggested by `help`, but rather overwrites existing file content